allo/backend/hls.py

```diff
@@ -314,9 +314,9 @@ def __call__(self, *args, shell=True):
             assert (
                 os.system(f"which {self.platform} >> /dev/null") == 0
             ), f"cannot find {self.platform} on system path"
-            ver = run_process("g++ --version", r"\d\.\d\.\d")[0].split(".")
+            ver = run_process("g++ --version", r"\d+\.\d+\.\d+")[0].split(".")
             assert (
-                int(ver[0]) * 10 + int(ver[1]) >= 48
+                int(ver[0]) * 4 + int(ver[1]) >= 48
```

Comment:
Maybe we can leave it as it is. `11*10 + 4>=48` also works for your case.

Reply:
What is the original minimum required g++ version?

My default Ubuntu 22.04 has version g++ 11.4.0. And 11*4+4=48.
And for g++ 12.x.y, 12*4+x >=48 also works. 
That's why I guessed the original factor should be 4, rather than 10.

Reply:
This file originated from an old project, so the minimal requirement is just 4.8. If you change the factor to 4, then `4 * 4 + 8 >= 48` does not work :)